### PR TITLE
feat(translation): add locale link +  translate

### PR DESCRIPTION
**Pull request by MatoPlus**
### Notion ticket link

[Support Locale Options for Backend APIs](https://www.notion.so/uwblueprintexecs/Support-Locale-Options-for-Backend-APIs-04ad8fd1292148838a42d3da5496f22f)
[Add Initial batch of translations for all supported languages](https://www.notion.so/uwblueprintexecs/Support-Locale-Options-for-Backend-APIs-04ad8fd1292148838a42d3da5496f22f)

https://user-images.githubusercontent.com/44826218/136678795-a59428b8-b28b-4301-8eb6-9dcff7f22c37.mp4

### Implementation description

- Remove locale options for API backends and use client side alternative (util helpers that filter translations based on current locale, this helps us save calls to db when the user changes local)
- Given the user locale, we know get the main translation based on the locale (on a call to a class/program, we grab ALL translations) and fallback to english if the translation for the current locale does not exists
- Add translations that we received from client
- Used i18n to do translations for class, program-details, and home page
- Enable date range and weekday translations by `toLocaleDateString()`
- Fix hrefs to use `next/link`, this preserves the locale when moving from pages to pages
- Add seed for translations

### Steps to test

1. Add translations to DB (class and program translations)
2. Create an accounts (volunteer/parent)
3. Test out translations in home page via the language modal
4. Test out translations in my classes page via the language modal
5. Test out translations in program details page via language modal

In all cases, translations should show up as expected. Note that we do not have all the translations we need, we will apply the form translations later when finalizing forms. 

### What should reviewers focus on?

- Dev: Kinda sketchy implementation of `weekdayToString` 👀 
- Dev: Link replacements `next/link` wrappers
- Design: Look and feel of translation/language modal flow
- Design: maybe try out the deployed instance!

### Checklist

-   [X] My PR name is descriptive and in imperative tense
-   [X] I have run the linter
-   [X] I have requested a review from the PL, and/or other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

**Review by bekahma — APPROVED** [review]
design approved :)

File: utils/weekdayToString.ts [hunk 1]
```diff
@@ -1,25 +1,48 @@
...
  * Method for returning the corresponding string given a Weekday enum
  * @param wd the Weekday enum value
+ * @param locale locale used
  */
-export default function weekdayToString(wd: weekday): string {
+export default function weekdayToString(wd: weekday, language: locale): string {
     switch (wd) {
         case weekday.MON:
-            return "Monday";
+            return moment()
```
Comment (feng-eric) [inline utils/weekdayToString.ts]: Seems like a lot of duplicate code. Any way we could reduce this?

File: src/components/WelcomeToSDC.tsx [hunk 2]
```diff
@@ -56,17 +51,20 @@ export const WelcomeToSDC: React.FC<WelcomeToSDCProps> = ({ session }) => {
...
-                                        onClick={goToLogin}
-                                        borderRadius={100}
-                                        fontWeight={"200"}
-                                    >
-                                        Register now
-                                    </Button>
+                                    <Link href="/login">
+                                        <Button
+                                            color="white"
+                                            backgroundColor="#0C53A0"
```
Comment (feng-eric) [inline src/components/WelcomeToSDC.tsx]: Can we make these constants?

File: src/components/WelcomeToSDC.tsx [hunk 3]
```diff
@@ -56,17 +51,20 @@ export const WelcomeToSDC: React.FC<WelcomeToSDCProps> = ({ session }) => {
...
+                                            _hover={{
+                                                backgroundColor: "#2C6AAD",
```
Comment (feng-eric) [inline src/components/WelcomeToSDC.tsx]: Same here

File: src/components/WelcomeToSDC.tsx [hunk 4]
```diff
@@ -89,17 +87,20 @@ export const WelcomeToSDC: React.FC<WelcomeToSDCProps> = ({ session }) => {
...
-                                        onClick={goToLogin}
-                                        borderRadius={100}
-                                        fontWeight={"200"}
-                                    >
-                                        Register now
-                                    </Button>
+                                    <Link href="/login">
+                                        <Button
+                                            color="white"
+                                            backgroundColor="#0C53A0"
```
Comment (feng-eric) [inline src/components/WelcomeToSDC.tsx]: Same as above

**Comment by MatoPlus** [comment]
Note to self, if PR #95, there might be some bugs in the mail method to fix in this PR (ie: need rebase).

File: src/components/WelcomeToSDC.tsx (hunk 2 above)
Comment (MatoPlus) [reply src/components/WelcomeToSDC.tsx]: smh classic color 🥴

File: utils/weekdayToString.ts [hunk 5]
```diff
@@ -1,25 +1,33 @@
-import { weekday } from "@prisma/client";
+import { locale, weekday } from "@prisma/client";
+import moment from "moment";
 
 /**
  * Method for returning the corresponding string given a Weekday enum
  * @param wd the Weekday enum value
+ * @param locale locale used
```
Comment (feng-eric) [inline utils/weekdayToString.ts]:
```suggestion
 * @param language locale used
```

**Review by feng-eric — APPROVED** [review]

**Pull request merged**